src/main/java/com/zufar/onlinestore/product/endpoint/ProductsEndpoint.java

```diff
@@ -29,9 +30,9 @@ public class ProductsEndpoint {
     private final ProductInfoDtoConverter productInfoDtoConverter;
 
     @GetMapping("/{id}")
-    public ResponseEntity<ProductInfoDto> getProductInfoById(@PathVariable("id") final String id) {
+    public ResponseEntity<ProductInfoDto> getProductInfoById(@PathVariable("id") final UUID id) {
         log.info("Received request to get the ProductInfo with id - {}.", id);
-        Optional<ProductInfo> ProductInfo = productInfoRepository.findById(Integer.parseInt(id));
+        Optional<ProductInfo> ProductInfo = productInfoRepository.findById(id);
```

Comment:
This is an old version of the Endpoint - you need to transfer it to the service (I'll do it)

Reply:
Yeah